File: docs/scripts/export_interface_contracts.py
#!/usr/bin/env python3
"""Export reviewed interface contracts verbatim for the self-hosted docs.

Run from any directory. --check verifies freshness without writing. These are
reference artifacts, not a declaration that the source contracts are complete.
"""
import argparse
from pathlib import Path

ROOT = Path(__file__).resolve().parents[2]
CONTRACTS = {
    "agent-server-openapi.json": "packages/agent-server/openapi.json",
}


def main():
    parser = argparse.ArgumentParser(description=__doc__)
    parser.add_argument("--check", action="store_true")
    args = parser.parse_args()
    for name, source in CONTRACTS.items():
        target = ROOT / "docs/public" / name
        content = (ROOT / source).read_bytes()
        if args.check:
            if not target.exists() or target.read_bytes() != content:
                raise SystemExit(f"Stale interface reference: {target}")
        else:
            target.write_bytes(content)
        print(f"{name}: {len(content)} bytes match {source}")


if __name__ == "__main__":
    main()

File: docs/scripts/ExportInterfaceContracts.hs
{-# LANGUAGE CPP #-}
module Main (main) where

import Control.Monad (unless)
import qualified Data.ByteString as ByteString
import System.Directory (canonicalizePath, doesFileExist)
import System.Environment (getArgs)
import System.Exit (die)
import System.FilePath ((</>), takeDirectory)

-- Export the canonical contract verbatim; --check never writes.
main :: IO ()
main = do
    args <- getArgs
    unless (args == [] || args == ["--check"]) $
        die "Usage: runghc docs/scripts/ExportInterfaceContracts.hs [--check]"
    root <- takeDirectory . takeDirectory . takeDirectory <$> canonicalizePath __FILE__
    let name = "agent-server-openapi.json"
        source = "packages/agent-server/openapi.json"
        target = root </> "docs/public" </> name
    content <- ByteString.readFile (root </> source)
    if args == ["--check"] then do
        exists <- doesFileExist target
        unless exists $ die ("Stale interface reference: " <> target)
        actual <- ByteString.readFile target
        unless (actual == content) $ die ("Stale interface reference: " <> target)
    else ByteString.writeFile target content
    putStrLn (name <> ": " <> show (ByteString.length content) <> " bytes match " <> source)
